osu.Game/Overlays/Profile/UserProfile.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Game.Online.API.Requests.Responses;
+
+namespace osu.Game.Overlays.Profile
+{
+    /// <summary>
+    /// Contains data about a profile presented on the <see cref="UserProfileOverlay"/>.
+    /// </summary>
+    public class UserProfile
```

Comment:
This name really confuses me. If the intention is to add an `Ruleset` context to an `APIUser`, I'd probably want this to be called `UserRulesetStats` or `UserStatsForRuleset` or even `UserForRuleset`. `UserProfile` sounds like it's either a drawable component, or a model with specific content for the profile display... but the end of the line for this class looks to just be an `APIUser` wrapper to add the ruleset.

Reply:
"Stats" feels weird since this wraps the whole `APIUser` model, not its statistics or otherwise. I would've gone with `UserProfileData`, or cut to the chase and went `APIUserWithRuleset`.

Reply:
/shrug

Pick one option and I'll change. I'm not attached to this name in the slightest.

Reply:
> `UserProfile` sounds like it's either a drawable component, or a model with specific content for the profile display...

I will say though that I would love this to eventually be the case because `APIUser` is horrible and I will continue saying it at every possible junction. It contains too much and half of its properties are invalid for reading in half of the context. It's not structured anything like what web has (which has `UserCompact`, then `User`, then also additional context-sensitive properties). and is also jammed into multiplayer innards and a multiplayer change is what made this PR required in the first place.

It's probably not going to happen though because the coupling is so high I'd have to spend a month on tedious PRs to rectify this.

Reply:
For sure. Our end goal was to get it to the point we can make it match. The only reason it's used in too many places right now is that before we were *subclassing* it to create "local" usages which was even worse (and not realm compatible). Definitely think when we have time for larger refactors this is an important one to address.

Reply:
As for naming, let's go with `UserProfileData`.

Reply:
Have applied the proposed name. Diff is probably quite bad now, and I'll probably have a grand old time resolving conflicts on the dependency, but discussions like these is why I split these in the first place.